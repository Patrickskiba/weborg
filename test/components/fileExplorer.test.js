import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
} from 'react-testing-library'
import 'jest-dom/extend-expect'
import indexedDB from 'idb-keyval'

jest.mock('../../src/utils/dropbox-files')

jest.mock('dropbox')

jest.mock('idb-keyval', () => ({
  get: jest.fn(() => new Promise(res => res('here is some text'))),
  keys: jest.fn(() => new Promise(res => res())),
}))

describe('fileExplorer tests', () => {
  afterEach(cleanup)
  it('renders all the file names stored in indededDB, clicking on a file changes it background and returns calls setText', async () => {
    indexedDB.keys.mockImplementationOnce(
      () => new Promise(res => res(['test1', 'test2', 'test3']))
    )
    const App = require('../../src/components/App').default
    const { getByTitle, getByText, getAllByText, container } = render(<App />)
    fireEvent.click(getByTitle('toggle-file-explorer'), { button: 1 })

    await waitForElement(() => getByText('test1'))

    expect(container).toMatchSnapshot()

    expect(getAllByText('Welcome to Web-org')[1]).toHaveClass(
      'makeStyles-highlighedText-99'
    )
    expect(getByText('test1')).toHaveClass('makeStyles-normalText-100')

    fireEvent.click(getByText('test1'), { button: 1 })

    fireEvent.click(getByTitle('toggle-file-explorer'), { button: 1 })

    await waitForElement(() => getByText('Welcome to Web-org'))

    expect(getByText('Welcome to Web-org')).toHaveClass(
      'makeStyles-normalText-100'
    )

    expect(getAllByText('test1')[1]).toHaveClass('makeStyles-highlighedText-99')
  })

  it('displays the default filename in the title area', () => {
    indexedDB.keys.mockImplementation(() => new Promise(res => res([])))
    const App = require('../../src/components/App').default
    const { getAllByTestId } = render(<App />)

    expect(getAllByTestId('filename-titlebar')[0]).toHaveTextContent(
      'Welcome to Web-org'
    )
  })
})
