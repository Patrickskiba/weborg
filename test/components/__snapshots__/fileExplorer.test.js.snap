// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileExplorer tests renders all the file names stored in indededDB, clicking on a file changes it background and returns calls setText 1`] = `
<div
  aria-hidden="true"
>
  <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            title="toggle-file-explorer"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
            data-testid="filename-titlebar"
          >
            Welcome to Web-org
          </h6>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            title="SettingsIcon"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </header>
    </div>
    <div
      class="sc-htoDjs juvQWM"
    >
      <div
        class="sc-dnqmqq eBtaWY"
      >
        <div>
          <div
            class="sc-bdVaJa iAUazu"
            data-testid="headline"
            style="color: black;"
          >
            <div
              class="sc-bwzfXH fANwYr"
            >
              <div
                class="sc-htpNat KtFsv"
              >
                <div
                  style="margin-right: 5px;"
                >
                  <svg
                    height="16px"
                    width="16px"
                  >
                    <circle
                      cx="50%"
                      cy="50%"
                      r="4px"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="sc-bxivhb eqfgSF"
              >
                <div>
                  <span
                    style="color: green; font-weight: 600;"
                  >
                     
                     
                  </span>
                  <span>
                    <span>
                      welcome
                    </span>
                     
                  </span>
                </div>
                <div>
                  <div
                    class="sc-EHOje eTPHNx"
                    style="color: rgb(113, 113, 113);"
                  >
                    <span>
                      <span>
                        test
                      </span>
                       
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="sc-ifAKCX hjKOjk"
              >
                <svg
                  height="16"
                  title="dash-collapse"
                  width="16"
                >
                   
                  <line
                    style="stroke: black; stroke-width: 1.5;"
                    x1="8"
                    x2="16"
                    y1="8"
                    y2="8"
                  />
                   
                </svg>
              </div>
            </div>
          </div>
        </div>
        <button
          aria-label="Add"
          class="MuiButtonBase-root MuiFab-root MuiFab-primary"
          style="position: fixed; right: 10px; bottom: 10px;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiFab-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              title="Add"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
